### ISSUE TITLE
Connector with file reader source does not honor the file type set to ASCII

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#532]
I'm trying to do my first steps in using mirth 1.6.1.2750 as JBoss MBean with a channel using a file reader connector. The files are "clear text" HL7 2.2 messages with ISO-8859-1 encoding.

Here is a sample:

MSH|^~&|SAP-ISH|DG|System1|Test|19991217100040||ADT^A04|15013015|P|2.2
EVN|A04|19991217100027
PID|||03234234^9||Smith^Charlotte^^^^||19200819|F|||Smith-Str. 161^^City^^82312^DE||03421-88734|||W|RK
PV1||I|ICU^^|||^^||^&&&&&&&&&&^^^^^|^&&&&&&&&&&^^^^^|||||^&&&&&&&&&&|||||0991039318^6|||||K|||||||||||||||IC|||||19991216080636|
ZPA|2|B^KK||19991217095933

In the dashboard I can see the the raw read message which looks like this:

TVNIfF5+XCZ8U0FQLUlTSHxER3xTeXN0ZW0xfFRlc3R8MTk5OTEyMTcxMDAwNDB8fEFEVF5BMDR8

MTUwMTMwMTV8UHwyLjIKRVZOfEEwNHwxOTk5MTIxNzEwMDAyNwpQSUR8fHwwMzIzNDIzNF45fHxT

bWl0aF5DaGFybG90dGVeXl5efHwxOTIwMDgxOXxGfHx8U21pdGgtU3RyLiAxNjFeXkNpdHleXjgy

MzEyXkRFfHwwMzQyMS04ODczNHx8fFd8UksKUFYxfHxJfElDVV5efHx8Xl58fF4mJiYmJiYmJiYm

Xl5eXl58XiYmJiYmJiYmJiZeXl5eXnx8fHx8XiYmJiYmJiYmJiZ8fHx8fDA5OTEwMzkzMTheNnx8

fHx8S3x8fHx8fHx8fHx8fHx8fElDfHx8fHwxOTk5MTIxNjA4MDYzNnwKWlBBfDJ8Ql5LS3x8MTk5

OTEyMTcwOTU5MzMK

So somehow the message is not read in as is.

The channel source is configured as follows:

File Type: ASCII
Encoding: ISO-8859-1

Obviously the file reader does not work as I thought it would. 

When running Mirth standalone this works with the same channel definition.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-524
Reporter: level420
Created: 2007-09-07T00:57:49.000-0700

**Comment by rbeckman-nextgen** [comment]
I've created a channel definition in Mirth 1.6.1 standalone and exported and imported the channel def. Mirth 1.6.1 MBean.

Now the file-reader is able to read the message as ASCII/ISO-8859-1 encoded without any problems.



Imported Comment. Original Details:
Author: level420
Created: 2007-09-17T05:25:10.000-0700

**Comment by rbeckman-nextgen** [comment]
Very interesting...can you post the channel that works and the one that doesn't work?

Thanks!

Imported Comment. Original Details:
Author: chrisl
Created: 2007-09-17T11:00:51.000-0700

**Comment by rbeckman-nextgen** [comment]
Sorry, but I'm not able to reproduce the bug anymore.

After reviewing the circumstances under which the problem showed up I think the problem was not the channel definition, but a problem with the channel deployment.

I think I first created the channel with the default settings for the file reader which is reading binary files.

I tested that and saw the error in the dashboard of Mirth administrator. Then I had a look into the "Raw Message" tab within the channel messages where I saw the binary hex data.

After that I corrected the settings in the file reader to "ASCII" and encoding "ISO-8859-1", saved the channel and re-deployed it. I started again to copy a HL7 file in to the file readers folder.

There was no change. The error in the dashboard showed up again.

I think the problem was that somehow the new settings in channel did not get saved or the modified channel did not get deployed.

As for now this bug should be put into the "could not reproduce" state.

Sorry for the traffic and wasting your time.






Imported Comment. Original Details:
Author: level420
Created: 2007-09-18T00:00:10.000-0700
